kitsune/products/models.py

```diff
@@ -169,6 +170,29 @@ def get_absolute_url(self):
             )
 
 
+class ProductTopic(ModelBase):
+    """Through model for Product and Topic to add additional metadata."""
+
+    product = models.ForeignKey(Product, on_delete=models.CASCADE)
+    topic = models.ForeignKey(Topic, on_delete=models.CASCADE)
+    created = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["product", "topic"], name="unique_product_topic")
+        ]
+
+
+class TopicSlugHistory(ModelBase):
+    topic = models.ForeignKey(Topic, on_delete=models.CASCADE, related_name="slug_history")
+    slug = models.SlugField(max_length=255, unique=True)
```

Comment:
Do we need to add `allow_unicode=True`? Hm, it appears that our topic slugs are never translated, so probably not.